From bb418b8b03e26214c220ee60cdbc8fb18337b33a Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 29 Feb 2012 10:41:39 +0900
Subject: [PATCH 03/97] rawparse: send the seek event without check if
 backward playback is set

This patch puts sending the seek event before check if backward
playback is set.
Backward playback could run if demuxers receive the seek event
such as qtdemux and so on.
---
 gst/rawparse/gstrawparse.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gst/rawparse/gstrawparse.c b/gst/rawparse/gstrawparse.c
index e4520f5..7cf1a33 100644
--- a/gst/rawparse/gstrawparse.c
+++ b/gst/rawparse/gstrawparse.c
@@ -697,15 +697,15 @@ gst_raw_parse_handle_seek_push (GstRawParse * rp, GstEvent * event)
   gst_event_parse_seek (event, &rate, &format, &flags, &start_type, &start,
       &stop_type, &stop);
 
-  /* can't seek backwards yet */
-  if (rate <= 0.0)
-    goto wrong_rate;
-
   /* First try if upstream handles the seek */
   ret = gst_pad_push_event (rp->sinkpad, event);
   if (ret)
     return ret;
 
+  /* can't seek backwards yet */
+  if (rate <= 0.0)
+    goto wrong_rate;
+
   /* Otherwise convert to bytes and push upstream */
   if (format == GST_FORMAT_TIME || format == GST_FORMAT_DEFAULT) {
     ret = gst_raw_parse_convert (rp, format, start, GST_FORMAT_BYTES, &start);
-- 
1.7.10.4

